cpg_utils/git.py

```diff
@@ -0,0 +1,185 @@
+
+
+def get_repo_name_from_remote(remote_name: str) -> str:
+    """
+    Get the name of a GitHub repo from a supported organization
+    based on its remote URL e.g.:
+    >>> get_repo_name_from_remote(\
+        'git@github.com:populationgenomics/cpg-utils.git'\
+    )
+    'cpg-utils'
+    >>> get_repo_name_from_remote(\
+        'https://github.com/populationgenomics/cpg-utils.git'\
+    )
+    'cpg-utils'
+
+    removed check for the authorised organisation(s) here - we receive
+    a full repository path, so we trust users will not attempt to run
+    potentially harmful code
+    """
+
+    repo = None
+    try:
+        if remote_name.startswith('http'):
+            match = re.match(r'https:\/\/[A-z0-9\.]+?\/.+?\/(?P<repo>.+)$', remote_name)
+            if match:
+                repo = match.group('repo')
+
+        elif remote_name.startswith('git@'):
+            match = re.match(r'git@[A-z0-9\.]+?:.+?\/(?P<repo>.+)$', remote_name)
+            if match:
+                repo = match.group('repo')
+    except AttributeError as ae:
+        raise Exception(f'Unsupported remote format: "{remote_name}"') from ae
+
+    if repo is None:
+        raise Exception(f'Unsupported remote format: "{remote_name}"')
+
+    if repo.endswith('.git'):
+        repo = repo[:-4]
+
+    return repo
+
+
+def check_if_commit_is_on_remote(commit: str) -> bool:
+    """
+    Returns 'True' if the commit is available on a remote branch.
+    This relies on the current environment to be up-to-date.
+    It asks if the local environment knows a remote branch with the commit.
+    """
+    command = ['git', 'branch', '-r', '--contains', commit]
+    try:
+        ret = subprocess.check_output(command)
+        return bool(ret)
+    except subprocess.CalledProcessError:
+        return False
+
+
+def prepare_git_job(
+    job,
+    organisation: str,
+    repo_name: str,
+    commit: str,
+    is_test: bool = True,
+    print_all_statements: bool = True,
+):
+    """
+    Takes a hail batch job, and:
+        * Clones the repository
+            * if access_level != "test": check the desired commit is on 'main'
+        * Check out the specific commit
+
+    Parameters
+    ----------
+    job                     - A hail BashJob
+    organisation            - The GitHub individual or organisation
+    repo_name               - The repository name to check out
+    commit                  - The commit hash to check out
+    is_test                 - CPG specific: only Main commits can run on Main data
+    print_all_statements    - logging toggle
+
+    Returns
+    -------
+    No return required
+    """
+
+    # Use "set -x" to print the commands for easier debugging.
+    if print_all_statements:
+        job.command('set -x')
+
+    # Note: for private GitHub repos we'd need to use a token to clone.
+    # Any job commands here are evaluated in a bash shell, so user arguments should
+    # be escaped to avoid command injection.
+    job.command(
```

Comment:
I don't think you'd want to quote the organisation and repo_name, just the full URL. Does GH even let you have repo names with characters you'd have to quote?